### ISSUE TITLE
Incorrect warning when trying to lint non-existing files

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1296]
I accidentally misspelled a file name and got a warning message that the file is being ignored because of my `.eslintignore` settings, which is not true.

**Example:**
Run `eslint /path/does/not/exist.js`
Output:

``` sh
/path/does/not/exist.js
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override

✖ 1 problem
```

If I run eslint with the `--no-ignore` option it outputs nothing, but I would still expect a warning.
